source/mir/math/stat.d

```diff
@@ -424,6 +465,286 @@ unittest
+    x[] = [1, 5, 3, 5, 2, 2];
+    assert(x.median!float.approxEqual(2.5));
+    x[] = [4, 3, 2, 1, 4, 5];
+    assert(x.median!float.approxEqual(3.5));
+    x[] = [4, 5, 3, 5, 5, 4];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [3, 3, 3, 0, 0, 1];
+    assert(x.median!float.approxEqual(2));
+    x[] = [4, 2, 2, 1, 2, 5];
+    assert(x.median!float.approxEqual(2));
+    x[] = [2, 3, 1, 4, 5, 5];
+    assert(x.median!float.approxEqual(3.5));
+    x[] = [1, 1, 4, 5, 5, 5];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [2, 4, 0, 5, 1, 0];
+    assert(x.median!float.approxEqual(1.5));
+    x[] = [3, 5, 2, 5, 4, 2];
+    assert(x.median!float.approxEqual(3.5));
+    x[] = [3, 5, 4, 1, 4, 3];
+    assert(x.median!float.approxEqual(3.5));
+    x[] = [4, 2, 0, 3, 1, 3];
+    assert(x.median!float.approxEqual(2.5));
+    x[] = [100, 4, 5, 0, 5, 1];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [100, 5, 4, 0, 5, 1];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [100, 5, 4, 0, 1, 5];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [4, 5, 100, 1, 5, 0];
+    assert(x.median!float.approxEqual(4.5));
+    x[] = [0, 1, 2, 2, 3, 4];
+    assert(x.median!float.approxEqual(2));
+    x[] = [0, 2, 2, 3, 4, 5];
+    assert(x.median!float.approxEqual(2.5));
+}
+
+version(mir_test)
+@safe pure nothrow
+unittest {
+    import mir.ndslice.slice: sliced;
+    import mir.math.common: approxEqual;
+
+    auto x0 = [9.0, 1, 0, 2, 3].sliced;
+    assert(x0.median.approxEqual(2));
+
+    auto x1 = [9.0, 1, 0, 2].sliced;
+    assert(x1.median.approxEqual(1.5));
+    
+    auto x2 = [9.0, 0, 1].sliced;
+    assert(x2.median.approxEqual(1));
+    
+    auto x3 = [1.0, 0].sliced;
+    assert(x3.median.approxEqual(0.5));
+    
+    auto x4 = [1.0].sliced;
+    assert(x4.median.approxEqual(1));
+}
+
+private pure @trusted nothrow @nogc
+F smallMedianImpl(F, Iterator, size_t N, SliceKind kind)(Slice!(Iterator, N, kind) slice) 
+{
+    size_t n = slice.elementCount;
+
+    assert(n > 0, "smallMedianImpl: slice must have elementCount greater than 0");
+    assert(n <= 5, "smallMedianImpl: slice must have elementCount of 5 or less");
+
+    import mir.ndslice.sorting: medianOf;
+    import mir.functional: naryFun;
+    import mir.utility: swapStars;
+
+    if (n > 2) {
+        auto sliceI0 = slice._iterator;
+        auto sliceI1 = sliceI0 + 1;
+        auto sliceI2 = sliceI1 + 1;
+        alias less = naryFun!("a < b");
+
+        if (n == 3) {
+            medianOf!less(sliceI0, sliceI1, sliceI2);
+            return cast(F) *sliceI1;
+        } else {
+            auto sliceI3 = sliceI2 + 1;
+            if (n == 4) {
+                // Put min in slice[0], lower median in slice[1]
+                medianOf!less(sliceI0, sliceI1, sliceI2, sliceI3);
+                // Ensure slice[2] < slice[3]
+                medianOf!less(sliceI2, sliceI3);
+                return mean!F(*sliceI1, *sliceI2);
```

Comment:
`0.5f * (*sliceI1 + *sliceI2)`

Reply:
If I do something like `smallMedianImpl!int(0, 2, 2, 3)`, then this change will cause the function to return a float. I assumed you preferred not to include the calls to `mean`, so I removed it and replaced it with `cast(F) (*sliceI1 + *sliceI2) / cast(F) 2;`. I also did something similar in other places.